addons/product/models/product_attribute.py

```diff
@@ -22,8 +22,28 @@ class ProductAttribute(models.Model):
         ('always', 'Always'),
         ('dynamic', 'Only when the product is added to a sales order')],
         default='always',
-        string="Create Variants",
-        help="Check this if you want to create multiple variants for this attribute.", required=True)
+        states={'used_on_product': [('readonly', True)]},
+        string="Variants Creation Mode",
+        help="This cannot be changed if the attribute is used on at least one product.", required=True)
+
+    state = fields.Selection(compute='_compute_state',
+        selection=[
+            ('not_used_on_product', "Not used on Product"),
+            ('used_on_product', "Used on Product")
+        ],
+        default='not_used_on_product',
```

Comment:
variants + computed fields is a bad Idea... this will lead for sure to performance problems, I estrongly reocmmend make an early optimization here making it store=True and/or do it with a simple write triggered from other models.

Reply:
It doesn't use products or variants to compute the state, it just requires a simple query more or less like "select attribute_id  from product_template_attribute_line where attribute_id in self.ids" and where attribute_id is indexed.
I don't think there there is going to be any performance issue with that.